platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/ProviderCodecs.kt

```diff
@@ -447,3 +449,30 @@ class MapPropertyCodec(
         }
     }
 }
+
+
+internal
+object GuardedDataCodec : Codec<GuardedData<*>> {
+    private
+    const val NO_OWNER = -1
+
+    override suspend fun WriteContext.encode(value: GuardedData<*>) {
```

Comment:
❓ Are `GuardedData` instances ever shared? In other words, should the codec care about their identities?

Reply:
It seems that could be the case with the newly introduced _shallow copies_

Reply:
It seems we care for identities for properties and configurable file collections, but not for providers, right? Guarded data are more akin to providers (and wrap one), right?

Do you see any problems?